### PR TITLE
Add elastic-charts deployment

### LINKED ISSUE #3: Shallow git clone
https://github.com/lukeelmers/terraform-kibana-dev/blob/1afb49002cb4603874816f3f24018523c16cb3b0/bootstrap.sh#L17-L19

Would it make sense to just clone the last commit of the branch of interest?

```bash
git clone --branch ${BRANCH} --single-branch --depth 1 "${REPO}.git" kibana
cd kibana
``` 

For a repo as large as Kibana, this should save some time on bootstrap.

### PR DESCRIPTION
**Pull request by kertal**
Closes #3

This PR enables elastic-charts deployment, so you can combine an elastic-charts PR with a Kibana PR. It works the following way:

`./kbn-dev.sh deploy pr {kibanaPR} --elastic-charts={elasticChartsPR}`

Contains some more fixes under the hood, added comments to them

### CODE REVIEW

File: gcp/kbn-dev.sh [hunk 1]
```diff
@@ -91,9 +100,16 @@ case $ACTION in
         log "🥬 Success deploying instance ${KIBANA_URL}"
         if [[ -n "$EUI" ]];
           then
-             log "🍉 Installing EUI"
+             log "🍅 Installing EUI"
```
Comment (kertal) [inline gcp/kbn-dev.sh:103]: There was too much Melon in this salad! More Tomate made it more balanced

File: gcp/scripts/check_server_online.sh [hunk 2]
```diff
@@ -1,13 +1,4 @@
...
-  if [[ $state == 'green' ]]
-  then
-    kibana_online=true
-  else
-    sleep 5
-  fi;
-done
\ No newline at end of file
+kibana_syslog=/var/log/syslog
+grep -m 1 "Kibana is now available" <(tail -f -n +1 $kibana_syslog)
```
Comment (kertal) [inline gcp/scripts/check_server_online.sh:4]: Due to an API change this did no longer work, so I decided to no longer depend on the API

File: gcp/scripts/install.sh [hunk 3]
```diff
@@ -9,8 +9,7 @@ BRANCH=$2
 sudo sysctl -w fs.inotify.max_user_watches=524288
 
 # clone kibana repo, check out branch, and configure environment
-git clone "$REPO.git"
-git -C ~/kibana checkout $BRANCH
+git clone --branch "${BRANCH}" --single-branch --depth 1 "${REPO}.git" kibana
```
Comment (kertal) [inline gcp/scripts/install.sh:12]: flat clone made it about 2 minutes faster, thx @jsanz for this hint
https://github.com/lukeelmers/terraform-kibana-dev/issues/3

File: gcp/kbn-dev.sh [hunk 4]
```diff
@@ -7,21 +7,42 @@ TYPE=$2
 VALUE=$3
 
 REPO_URL=${REPO_URL:-"https://github.com/elastic/kibana"}
+
+GCP_NAME_PREFIX=${KBN_GCP_PREFIX:-"kbn-dev-v1"}
```
Comment (kertal) [inline gcp/kbn-dev.sh:11]: now the usual prefix is configureable so you can 

`export KBN_GCP_PREFIX=my-nice-gcp-name-prefix`

to use a different prefix in GCP

**Review by lukeelmers — COMMENTED** [review]
Thanks @kertal, this is a great addition! Tested and works as expected.

Would it be possible to update the README & the CLI `help` text with this info? Otherwise LGTM

File: gcp/kbn-dev.sh (hunk 1 above)
Comment (kertal) [reply gcp/kbn-dev.sh:103]: Glad the chef gourmet of this repo approves :)

File: gcp/kbn-dev.sh [hunk 5]
```diff
@@ -91,9 +100,16 @@ case $ACTION in
...
         if [[ -n "$EUI" ]];
           then
-             log "🍉 Installing EUI"
+             log "🍅 Installing EUI"
              eval "ssh -q -o StrictHostKeyChecking=no ubuntu@${PUBLIC_IP} /tmp/install_eui.sh ${EUI}"
         fi
+
+        if [[ -n "$ELASTIC_CHARTS" ]];
+          then
+             log "🥒 Installing Elastic Charts"
```
[the remark this answers is not in the document]
Comment (kertal) [reply gcp/kbn-dev.sh:109]: sure added more info! thx

**Review by lukeelmers — APPROVED** [review]
LGTM, thanks again for taking this to the next level! 💪

**Pull request merged**